circle.yml

```diff
@@ -6,12 +6,24 @@ jobs:
       - image: rust:1
     steps:
       - checkout
+      - restore_cache:
+          key: project-cache
       - run:
           name: Update rustc
           command: |
             rustup target add wasm32-unknown-unknown
+            rustup component add rustfmt-preview
             rustup update
+      - run:
+          name: Check formatting
+          command: |
+            rustfmt --version
+            cargo fmt
+            git diff --exit-code
       - run:
           name: Build
           command: cargo build --release --target wasm32-unknown-unknown
-
+      - save_cache:
+          key: project-cache
+          paths:
+            - "~/.cargo"
```

Comment:
Add the path `./target` as this is where the binaries for the project and its dependencies are stored.